org.mfek.glif.yml

```diff
@@ -0,0 +1,75 @@
+app-id: org.mfek.glif
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+cleanup: 
+  - /sdk
+build-options:
+  append-path: /app/sdk/rust-nightly/bin
+  env: 
+    CARGO_HOME: /run/build/MFEKglif/cargo
+    SKIA_BINARIES_URL: "file:///run/build/MFEKglif/skia-binaries.tar.gz"
+    FORCE_SKIA_BINARIES_DOWNLOAD: "1"
+finish-args: 
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --device=dri
+  - --filesystem=home
+command: MFEKglif
+modules:
+  - name: MFEKglif
+    buildsystem: simple
+    build-commands:
+      - cd rust && ./install.sh --prefix=/app/sdk/rust-nightly --without=rust-docs
+        --without=rust-docs-json-preview --disable-ldconfig --verbose
+      - cp ./resources/MFEKglif.desktop ./${FLATPAK_ID}.desktop
+      - cp ./resources/MFEKglif.png ./${FLATPAK_ID}.png
+      - desktop-file-edit ./${FLATPAK_ID}.desktop --set-key=Icon --set-value=org.mfek.glif.png
+      - cargo --offline fetch --manifest-path Cargo.toml --verbose
+      - cargo --offline build --release --features=sdl2-static
+      - install -Dm755 ./target/release/MFEKglif -t /app/bin/
+      - install -Dm644 ./${FLATPAK_ID}.metainfo.xml -t /app/share/metainfo/
+      - install -Dm644 ./${FLATPAK_ID}.desktop -t /app/share/applications/
+      - install -Dm644 ./${FLATPAK_ID}.png -t /app/share/icons/hicolor/512x512/apps/
+    sources:
+      - type: git
+        url: https://github.com/MFEK/glif.git
+        commit: 8764b3e40a6dc219f3dc7db1429c58dff9795840
+      - type: file
+        only-arches:
+          - x86_64
+        url: https://github.com/rust-skia/skia-binaries/releases/download/0.62.0/skia-binaries-8cf4841aefdb295709e9-x86_64-unknown-linux-gnu-gl.tar.gz
+        dest-filename: skia-binaries.tar.gz
+        sha256: e4806ae8566b1cc1e7692d9c5f0e0a32754a07591db6de884c107adea1922b7a
+      - type: file
+        only-arches:
+          - aarch64
+        url: https://github.com/rust-skia/skia-binaries/releases/download/0.62.0/skia-binaries-8cf4841aefdb295709e9-aarch64-unknown-linux-gnu-gl.tar.gz
+        dest-filename: skia-binaries.tar.gz
+        sha256: e4806ae8566b1cc1e7692d9c5f0e0a32754a07591db6de884c107adea1922b7a
```

Comment:
the same sha256 for both arch? chances are this isn't right.

Reply:
Yes this was a mistake. I hastily copy pasted and then didn't realise until seeing the build fail that the file didn't even exist. There is no distribution for aarch64 gnu linux. The reason I was using this file in the first place was that building from source for this specific library proved infeasible offline

Reply:
Yes, that wouldn't be right. I'm trying a real aarch64 build shortly as I have an aarch64 Arch VM.

First I am applying patches to make writing .desktop and icon unneeded, MFEKglif can now do it at runtime.

![Screenshot_20231129_013700](https://github.com/flathub/flathub/assets/838783/d0fc22f5-468e-4cfb-aefa-3743a9ea8a7d)

Reply:
MFEKglif opens /dev/shm/MFEKglif, copies into it the icon and .desktop.
It informs ipc.rlib of this, which creates symlinks from $XDG_SHARE_DIR/{applications,icons} to /dev/shm.
This way, MFEKglif can hopefully soon support showing the current glyph being edited as the icon, like FontForge does.
Invariably, this breaks some best practice, like the rest of the project. :o)